homework9/task3/file_counter.py

```diff
@@ -0,0 +1,48 @@
+"""
+Write a function that takes directory path, a file extension and an optional tokenizer.
+It will count lines in all files with that extension if there are no tokenizer.
+If a the tokenizer is not none, it will count tokens.
+For dir with two files from hw1.py:
+#>>> universal_file_counter(test_dir, "txt")
+6
+#>>> universal_file_counter(test_dir, "txt", str.split)
+6
+"""
+import glob
+import os
+from pathlib import Path
+from typing import Callable, Optional, TextIO
+
+
+def tokenizer_generator(file_handler: TextIO, tok: Optional[Callable]):
+    buffer = ""
+    char = " "
+    while char:
+        char = file_handler.read(1)
+        if not char:
+            yield 1  # buffer
```

Comment:
this will lead to an extra "1" in counter value, isn't it?

Reply:
no, there is stay last token in buffer after file is ending